src/Microsoft.Azure.WebJobs.Host/Scale/IConcurrencyStatusRepository.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Provides functionality for <see cref="HostConcurrencySnapshot"/> persistence.
+    /// </summary>
+    public interface IConcurrencyStatusRepository
+    {
+        /// <summary>
+        /// Writes the specified snapshot.
+        /// </summary>
+        /// <param name="snapshot">The snapshot to persist.</param>
+        /// <returns>A task that completes when the write is finished.</returns>
+        Task WriteAsync(HostConcurrencySnapshot snapshot);
+
+        /// <summary>
+        /// Reads the last host concurrency snapshot if present.
+        /// </summary>
+        /// <returns>A task that returns the snapshot if present, or null.</returns>
+        Task<HostConcurrencySnapshot?> ReadAsync();
```

Comment:
Using nullable reference type annotations here and a few other places. You'll see that many of the files using these interfaces have the `#nullable enable` pragma to leverage the feature.